### DISCUSSION TITLE
Self signed certificate for AIO Dashboard unnacceptable in firefox

### DISCUSSION THREAD

**Discussion by navyjosh** [nextcloud/all-in-one#3411] [Questions] [answered]
### Is your feature request related to a problem? Yes.

I cannot access aio dashboard using my default browser (Firefox v117.0.1) on Ubuntu 22.04.

When I click the _Open Nextcloud AIO Interface_ button firefox tells me self signed certificate is not acceptable and there is no proceed button.  I tried adding the nextcloud.local cert to my  browser's CA but that still didn't work.

I am using Traefik v2 as a reverse proxy in front Nextcloud AIO.

### Describe the solution you'd like
More documentation in the reverse proxy config setup on how to configure Traefik v2 to serve valid certificates for my domain and proxy properly to AIO dashboard. I would like to use a subdomain like aio.mydomain.com to proxy to the AIO dashboard.

Alternatively, will this solve the issue? https://stackoverflow.com/a/77009337

**Comment by szaimen** [comment]
Hi, can you post a screemshot of the error in firefox?

**Reply by navyjosh** [reply]
![image](https://github.com/nextcloud/all-in-one/assets/6722547/880ec8b8-5b2f-4ea2-8cc4-5679cf3e565c)

**Reply by Zoey2936** [reply]
try to use the ip, without a domain

**Reply by navyjosh** [reply]
![image](https://github.com/nextcloud/all-in-one/assets/6722547/bbfecd33-53ea-49d4-bd42-acffbef68e65)

**Reply by navyjosh** [reply]
Using the IP was the same error. But if I use another browser, then I can bypass security issue and go in.

**Reply by Zoey2936** [reply]
your domain has with a high chance HSTS enabled, which makes it need of a valid TLS certificate, so to fix this, just use your servers domain, that doesn't use HSTS

**Reply by navyjosh** [reply]
I have a valid TLS cert, is there a way to configure the AIO dashboard to use the cert?

**Reply by Zoey2936** [reply]
If you run AIO behind a reverse proxy, you can try to run the AIO dashboard also behind it

**Reply by navyjosh** [reply]
Also, I can access the dashboard just fine from another browser, chromium for example:
![image](https://github.com/nextcloud/all-in-one/assets/6722547/1374a1d4-8b33-425e-a010-82120d74a254)

**Reply by navyjosh** [reply]
I am running it behind Traefik v2, I couldn't figure out how to configure it properly. Does the Dashboard run behind the apache server or is the AIO Dashboard in front of the apache server?

**Reply by navyjosh** [reply]
In order to run it behind my reverse proxy I would need the AIO dashboard to run on http instead of https. I could not find documentation that covers this

**Reply by Zoey2936** [reply]
Does Traefik not support proxying to https?

**Reply by navyjosh** [reply]
I imagine it does, wouldn't that be additional, unnecessary overhead though? I guess on the admin dashboard we won't be need to access it very frequently and won't have a huge amount of requests so the additional overhead might not matter too much.

**Reply by Zoey2936** [reply, marked as answer]
if traefik does support it, just use it

**Reply by mrjovanovic** [reply]
For anyone Googling this and looking for an answer, while using nginx-proxy: you want to set up proxy forwarding from HTTPS to an HTTPS upstream. Something like this works for me.  
```
 server {
                listen 443 ssl;
                server_name aio.yourdomain.org;
                http2 on;

                location / {
                        proxy_pass https://UPSTREAM_IP:8080$request_uri;
                        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                        proxy_set_header X-Forwarded-Port $server_port;
                        proxy_set_header X-Forwarded-Scheme $scheme;
                        proxy_set_header X-Forwarded-Proto $scheme;
                        proxy_set_header X-Real-IP $remote_addr;
                        proxy_set_header Accept-Encoding "";
                        proxy_set_header Host $host;
                        proxy_ssl_server_name on;

                        proxy_http_version 1.1;
                        proxy_set_header Upgrade $http_upgrade;
                        proxy_set_header Connection "upgrade";
                }
        }
```

Obviously add your SSL parameters there, as well as your own information. Note the http**s** in the `proxy_pass` directive, as well remember to set `proxy_ssl_server_name on`. 

You will have to stop aio-apache first with `docker stop nextcloud-aio-apache`, log in with the password for AIO you first set up, and then restart your containers. Make a backup first. You will be able to access it from the administration settings easily after that.
